daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1071,24 +1070,32 @@ private[lf] object SBuiltin {
           }
           machine.returnValue = cached.any
         case None =>
-          throw SpeedyHungry(
-            SResultNeedContract(
-              coid,
-              onLedger.committers,
-              { case Versioned(_, V.ContractInstance(actualTmplId, arg, _)) =>
-                machine.pushKont(KCacheContract(machine, coid))
-                machine.ctrl = SEApp(
-                  // The call to ToCachedContractDefRef(actualTmplId) will query package
-                  // of actualTmplId if not know.
-                  SEVal(ToCachedContractDefRef(actualTmplId)),
-                  Array(
-                    SEImportValue(Ast.TTyCon(actualTmplId), arg),
-                    SEValue(args.get(1)),
-                  ),
-                )
-              },
+          def continue = { case V.ContractInstance(actualTmplId, arg, _) =>
```

Comment:
if you want to simplify things in Speedy, consider having `ResultNeedContract` keep a versioned contract instance but only drop it from `SResultNeedContract`. Then you don’t break Canton & the participant.

Reply:
Ok, I did that.